internal/user_access_test.go

```diff
@@ -0,0 +1,101 @@
+package internal
+
+import (
```

Comment:
Thanks for adding some testing in here! 🥇 

For a future discussion: these tests are likely going to change away from being pure unit tests once there's a database microservice to talk to. We'll want to pay mind to how these tests may evolve if they move away from being pure unit tests to tests that rely on a separate component.

Reply:
💯: Unit -> integration -> functional-> p2p